lib/rouge/lexers/batch.rb

```diff
@@ -0,0 +1,165 @@
+        cls color copy date del erase dir dpath
+        echo ftype move pause path prompt
+        setlocal endlocal shift start
+        time title type ren rename
+        ver verify vol
+      ).join('|')
+
+      ATTRIBUTES = %w(
+        on off disable
+        enableextensions enabledelayedexpansion
+      ).join('|')
+
+      EXTERNAL = %w(
+        addusers admodcmd ansicon arp at attrib
+        bcdboot bcdedit bitsadmin browstat
+        cacls certreq certutil
+        change chcp chkdsk chkntfs choice
+        cidiag cipher cleanmgr clip cmd cmdkey
+        comp compact compress convert convertcp coreinfo csccmd csvde
+        cscript curl
+        debug defrag delprof deltree devcon diamond dirquota diruse
+        diskpart diskshadow diskuse dism dnscmd doskey driverquery
+        dsacls dsadd dsget dsquery dsmod dsmove dsrm dsmgmt dsregcmd
+        edlin eventcreate expand extract
+        fc fdisk find findstr fltmc forfiles format freedisk fsutil ftp
+        getmac gpresult gpupdate help hostname
+        icacls ifmember inuse ipconfig kill
+        label lgpo lodctr logman logoff logtime
+        makecab mapisend mbsacli mem
+        mklink mode more mountvol moveuser msg mshta msiexec msinfo32 mstsc
+        nbtstat net net1 netdom netsh netstat nlsinfo nltest now nslookup
+        ntbackup ntdsutil ntoskrnl ntrights nvspbind
+        openfiles
+        pathping perms ping popd portqry powercfg pngout
+        pnputil print printbrm prncnfg prnmngr procdump
+        psexec psfile psgetsid psinfo pskill pslist
+        psloggedon psloglist pspasswd psping
+        psservice psshutdown pssuspend
+        qbasic qgrep qprocess query quser qwinsta
+        rasdial recover reg reg1 regdump regedt32
+        regsvr32 regini replace reset restore
+        rundll32 rmtshare robocopy route rpcping run runas
+        sc scandisk schtasks setspn setx sfc
+        share shellrunas shortcut shutdown
+        sigcheck sleep slmgr sort strings subinacl
+        subst sysmon systeminfo
+        takeown taskkill tasklist telnet tftp
+        timeout tlist touch tracerpt tracert tree tscon
+        tsdiscon tskill tttracer typeperf tzutil
+        undelete unformat
+        verifier vmconnect vssadmin
+        w32tm waitfor wbadmin wecutil wevtutil wget
+        where whoami windiff winrm winrs wmic wpeutil wpr wusa wuauclt
+        wscript
+        xcopy
+      ).join('|')
+
+      state :basic do
+        # Comments
+        rule %r/\b(REM)\b.*$/i, Comment
+        # Empty Labels
+        rule %r/^::.*$/, Comment
+
+        # Labels
+        rule %r/:[a-z]+/i, Keyword::Constant
+
+        # Devices
+        rule %r/@?\b(#{DEVICES})\s*\b/i, Keyword::Reserved
+        rule %r/@?\b(#{DEVICESNUM}[0-9])\s*\b/i, Keyword::Reserved
+
+        # Lang Keywords
+        rule %r/@?\b(#{KEYWORDS})\s*\b/i, Name::Builtin::Pseudo
+        # Internal Commands
+        rule %r/@?\b(#{INTERNAL})\s*\b/i, Name::Builtin::Pseudo
```

Comment:
Is there an example of this one in the visual sample?

Reply:
Yes, for example: `@echo off`